cranelift-codegen/meta/src/cdsl/instructions.rs

```diff
@@ -172,30 +203,11 @@ impl Instruction {
             None => Vec::new(),
         }
     }
+}
 
-    pub fn bind(&self, lane_type: impl Into<LaneType>) -> BoundInstruction {
-        bind(self.clone(), Some(lane_type.into()), Vec::new())
-    }
-
-    pub fn bind_ref(&self, reference_type: impl Into<ReferenceType>) -> BoundInstruction {
-        bind_ref(self.clone(), Some(reference_type.into()), Vec::new())
-    }
-
-    pub fn bind_vector_from_lane(
-        &self,
-        lane_type: impl Into<LaneType>,
-        vector_size_in_bits: u64,
-    ) -> BoundInstruction {
-        bind_vector(
-            self.clone(),
-            lane_type.into(),
-            vector_size_in_bits,
-            Vec::new(),
-        )
-    }
-
-    pub fn bind_any(&self) -> BoundInstruction {
-        bind(self.clone(), None, Vec::new())
+impl Bindable for Instruction {
```

Comment:
I've been going back and forth about this: does it make sense to have a `BoundInstruction` that isn't bound? `my_inst.bind()` is a valid `BoundInstruction` but it isn't really bound to anything yet. I worried that adding `BoundInstruction::new` would make people think this was ok but I kept that method private to limit the confusion to only this one file. I would prefer to code this so that `Instruction` and `InstSpec` can only become a `BoundInstruction` when they are truly bound but I could be convinced otherwise.

Reply:
You're right it might create some confusion when finishing building encodings in particular; I was trying to avoid the trait altogether, the (meta crate) compile cost that comes with it and the need to have use statements in all the files that use it. The current way looks good, thanks.